### PR TITLE
[Merged by Bors] - chore: backports for leanprover/lean4#4814 (part 25)

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/WittVector/Teichmuller.lean [hunk 1]
```diff
@@ -70,14 +70,14 @@ private theorem map_teichmullerFun (f : R →+* S) (r : R) :
   · rfl
   · exact f.map_zero
 
-private theorem teichmuller_mul_aux₁ (x y : MvPolynomial R ℚ) :
+private theorem teichmuller_mul_aux₁ {R} (x y : MvPolynomial R ℚ) :
```
Comment (sgouezel) [inline Mathlib/RingTheory/WittVector/Teichmuller.lean:73]:
```suggestion
private theorem teichmuller_mul_aux₁ {R : Type*} (x y : MvPolynomial R ℚ) :
```

File: Mathlib/RingTheory/WittVector/Teichmuller.lean [hunk 2]
```diff
@@ -70,14 +70,14 @@ private theorem map_teichmullerFun (f : R →+* S) (r : R) :
...
     teichmullerFun p (x * y) = teichmullerFun p x * teichmullerFun p y := by
   apply (ghostMap.bijective_of_invertible p (MvPolynomial R ℚ)).1
   rw [RingHom.map_mul]
   ext1 n
   simp only [Pi.mul_apply, ghostMap_apply, ghostComponent_teichmullerFun, mul_pow]
 
-private theorem teichmuller_mul_aux₂ (x y : MvPolynomial R ℤ) :
+private theorem teichmuller_mul_aux₂ {R} (x y : MvPolynomial R ℤ) :
```
Comment (sgouezel) [inline Mathlib/RingTheory/WittVector/Teichmuller.lean:80]:
```suggestion
private theorem teichmuller_mul_aux₂ {R : Type*} (x y : MvPolynomial R ℤ) :
```

File: Mathlib/RingTheory/WittVector/Verschiebung.lean [hunk 3]
```diff
@@ -49,13 +49,13 @@ theorem verschiebungFun_coeff_succ (x : 𝕎 R) (n : ℕ) :
   rfl
 
 @[ghost_simps]
-theorem ghostComponent_zero_verschiebungFun (x : 𝕎 R) :
+theorem ghostComponent_zero_verschiebungFun [hp : Fact p.Prime]  (x : 𝕎 R) :
```
Comment (sgouezel) [inline Mathlib/RingTheory/WittVector/Verschiebung.lean:52]:
```suggestion
theorem ghostComponent_zero_verschiebungFun [hp : Fact p.Prime] (x : 𝕎 R) :
```

File: Mathlib/RingTheory/WittVector/Verschiebung.lean [hunk 4]
```diff
@@ -49,13 +49,13 @@ theorem verschiebungFun_coeff_succ (x : 𝕎 R) (n : ℕ) :
...
     ghostComponent 0 (verschiebungFun x) = 0 := by
   rw [ghostComponent_apply, aeval_wittPolynomial, Finset.range_one, Finset.sum_singleton,
     verschiebungFun_coeff_zero, pow_zero, pow_zero, pow_one, one_mul]
 
 @[ghost_simps]
-theorem ghostComponent_verschiebungFun (x : 𝕎 R) (n : ℕ) :
+theorem ghostComponent_verschiebungFun [hp : Fact p.Prime]  (x : 𝕎 R) (n : ℕ) :
```
Comment (sgouezel) [inline Mathlib/RingTheory/WittVector/Verschiebung.lean:58]:
```suggestion
theorem ghostComponent_verschiebungFun [hp : Fact p.Prime] (x : 𝕎 R) (n : ℕ) :
```

**Review by sgouezel — COMMENTED** [review]
bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**